core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1133,7 +1133,9 @@ class Log(@volatile var dir: File,
       val fetchDataInfo = readUncommitted(offset, 1)
       fetchDataInfo.fetchOffsetMetadata
     } catch {
-      case _: OffsetOutOfRangeException => LogOffsetMetadata.UnknownOffsetMetadata
+      case _: OffsetOutOfRangeException =>
+        val firstOffset = segments.firstEntry.getValue.baseOffset
```

Comment:
Seems a little surprising to return the first offset in the log if the searched offset couldn't be found. I wonder if it would be clearer to change this function to return `Option[LogOffsetMetadata]` and to handle the `None` case in `convertHWToLocalOffsetMetadata` by converting to the start offset?

Reply:
Thanks for the review! Sure. It is fixed as suggested.